torch/csrc/jit/tracer.h

```diff
@@ -163,6 +158,23 @@ inline Value* getValueTrace(const std::shared_ptr<TracingState>& state, const Va
   return constant;
 }
 
+inline Value* getOutputTrace(const std::shared_ptr<TracingState>& state, const Variable& var, size_t output_no) {
+  if (!var.defined()) {
+    Node *n = state->graph->createUndefined();
+    return state->graph->appendNode(n)->output();
+  }
+
+  auto vts = detail::getValueState(state, var, false);
+  if (!vts) {
+    std::ostringstream os;
+    os << "output " << output_no << " of traced region did not have observable "
+       << "data dependence with trace inputs; this probably indicates your program "
+       << "cannot be understood by the tracer.  Contents of variable:\n" << var;
```

Comment:
On the contrary, while I was debugging this problem, as soon as I saw the contents of the variable I knew *exactly* what the problem was (it was zeros with a stripe of ones, and that screamed "inplace view update!") Since this error is a strictly compiler internal error, I would err on the side of more debug information.

Reply:
Does ATen only print summary of a tensor like PyTorch, or are these full contents? If the second, then I disagree. Having to scroll a few thousand lines up every time you’re trying to debug this error is not what you want (especially if you’re a user and hit this accidentally).

Reply:
If it is the second, that is surely an ATen bug to be fixed.

Reply:
Still, I don't want to see huge tensors in bug reports, because that doesn't carry any info for us. If you want to see what these tensors are because you don't know, you can always add the prints yourself.